python_modules/dagster/dagster/_core/execution/context/compute.py

```diff
@@ -1351,12 +1352,76 @@ def get() -> "OpExecutionContext":
         ctx = _current_asset_execution_context.get()
         if ctx is None:
             raise DagsterInvariantViolationError("No current OpExecutionContext in scope.")
-        return ctx.get_op_execution_context()
+        return ctx.op_execution_context
+
+
+###############################
+######## AssetExecutionContext
+###############################
+
+ALTERNATE_METHODS = {
+    "run_id": "run_properties.run_id",
+    "run": "run_properties.dagster_run",
+    "dagster_run": "run_properties.dagster_run",
+    "run_config": "run_properties.run_config",
+    "retry_number": "run_properties.retry_number",
+}
+
+
+def _get_deprecation_kwargs(attr: str):
+    deprecation_kwargs = {"breaking_version": "1.8.0"}
+    deprecation_kwargs["subject"] = f"AssetExecutionContext.{attr}"
+
+    if attr in ALTERNATE_METHODS:
+        deprecation_kwargs["additional_warn_text"] = (
+            f"You have called the deprecated method {attr} on AssetExecutionContext. Use"
+            f" context.{ALTERNATE_METHODS[attr]} instead."
+        )
+
+    return deprecation_kwargs
+
+
+class RunProperties(
```

Comment:
Naming? Other options discussed in [this thread](https://dagsterlabs.slack.com/archives/C047L6H0LF4/p1698339635135189):
* `ContextRunProperties` (`ContextAssetProperties`, `ContextPartitionProperties`)
* `ExecutionRunProperties` (`ExecutionAssetProperties`, `ExecutionPartitionProperties`)
* `RunDetails` (`AssetDetails`, `PartitionDetails`)